[material-ui][Input] Remove unused classes from InputClasses interface

sai6855 opened the pull request:
<!-- Thanks so much for your PR, your contribution is appreciated! ❤️ -->

- [ ] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/material-ui/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

sai6855 commented on packages/mui-material/src/Input/inputClasses.ts:
```diff
@@ -5,14 +5,10 @@ import { inputBaseClasses } from '../InputBase';
   /** Styles applied to the root element if `disabled={true}`. */
   disabled: string;
-  /** Styles applied to the root element if color secondary. */
-  colorSecondary: string;
```
Instead of adding the class to the DOM, I chose to remove it here. The fact that no users have reported this issue, even though it has likely existed for a long time (possibly since v5), suggests that it is not widely used. Therefore, I decided to remove the class instead of adding it to the DOM.

ZeeshanTamboli reviewed (commented):
@sai6855 I see there are several unused classes in the `Input` component, such as `sizeSmall`, `inputSizeSmall`, `multiline`, `inputMultiline`, `inputAdornedStart` and others. Only these classes are actually used: `root`, `focused`, `error`, `disabled`, `underline`, and `input`.

sai6855 commented:
> @sai6855 I see there are several unused classes in the `Input` component, such as `sizeSmall`, `inputSizeSmall`, `multiline`, `inputMultiline`, `inputAdornedStart` and others. Only these classes are actually used: `root`, `focused`, `error`, `disabled`, `underline`, and `input`.

Yeah you are right, removed all usused classes and added tests to verify same

sai6855 commented:
@ZeeshanTamboli i agree it's a breaking change if classes are removed, what do you think of last 2 commits? it keeps class in interface but removes info from docs so that users cannot see wrong info on classes

ZeeshanTamboli commented:
@sai6855 If those class names are still used for styling, why remove them from the docs?

sai6855 commented:
> @sai6855 If those class names are still used for styling, why remove them from the docs?

take this class for example https://mui.com/material-ui/api/input/#input-classes-MuiInput-colorSecondary, it's not used anywhere and this class does't end up in DOM. so this is wrong info know?

ZeeshanTamboli commented:
> but it doesn't work 

It does work. Check this example: https://codesandbox.io/p/sandbox/purple-violet-sgk8hf. Type in the input, and you'll see the text color is red. Although the DOM shows the class as `MuiInputBase-colorSecondary`, we're using `inputClasses.secondary` (`MuiInput-colorSecondary`).

ZeeshanTamboli commented:
@sai6855 Ah, I see now. I was confused about how `inputClasses.secondary` targeted `MuiInputBase-colorSecondary`. It turns out `inputClasses.secondary` maps to `MuiInputBase-colorSecondary`, not `MuiInput-colorSecondary`, which isn’t applied in the DOM.  

So, it makes sense to remove it from the docs. Alternatively, we could mark it as deprecated since developers are already using it. Either way, we should get a review from @DiegoAndai on this.

sai6855 commented:
@DiegoAndai Here's the summary of above discussion

There are several classes in `Input` classes API documentaion that doesn't work because these classes doesn't end up in DOM.
One such example is this https://mui.com/material-ui/api/input/#input-classes-MuiInput-colorSecondary

proof for `.MuiInput-colorSecondary` is not working: https://github.com/mui/material-ui/pull/44987/files#diff-a91d61a487c6fd5c001cd5cf45a1ab5393cdcd9c0efaf87c72b9b1eb5b3eefd2R54

This PR aims to remove any information regarding unused classes, we could try 2 ways to solve this

1. Remove `colorSecondary` from `InputClasses` interface  https://github.com/mui/material-ui/blob/0221d17856a63d89e832d49f652b1e189ab7b7cf/packages/mui-material/src/Input/inputClasses.ts#L14-L15, but this would be breaking change for users who are using `inputClasses.colorSecondary`
2. Add `@ignore` in classes `JSDoc` which removes class related info from docs (Right now, i've implemented this)

which approach do you suggest?

siriwatknp commented:
> 1. but this would be breaking change for users who are using `inputClasses.colorSecondary`

I think if it's a breaking change. It should be done in v7 branch (I think @DiegoAndai will create the branch soon).
